FrameworkSystem/Client/BundleDeliveryClient.py

```diff
@@ -49,8 +70,19 @@ def __setHash(self, bundleID, dirToSyncTo, bdHash):
       self.log.error("Could not save hash after synchronization", "%s: %s" % (fileName, str(e)))
 
   def syncDir(self, bundleID, dirToSyncTo):
+    """ Synchronize directory
+
+        :param str bundleID: bundle ID
+        :param str dirToSyncTo: path to sync directory
+
+        :return: S_OK(bool)/S_ERROR()
+    """
     dirCreated = False
-    if not os.path.isdir(dirToSyncTo):
+    if so.path.isdir(dirToSyncTo):
+      for p in [os.W_OK, os.R_OK]:
```

Comment:
I think I prefer the original approach with catching the exception many file systems don't respect the unix permission bits. The [Python docs](https://docs.python.org/3/library/os.html#os.access) warn about this:

> Note: I/O operations may fail even when `access()` indicates that they would succeed, particularly for operations on network filesystems which may have permissions semantics beyond the usual POSIX permission-bit model.

Reply:
It seems to me that this check will not be superfluous.. 
In addition I'll add catching OSError exception for ``tF.extract(tarinfo, dirToSyncTo)``